### PR TITLE
Disables the submit button for empty values in tickets form

### LINKED ISSUE #3358: Ticket is generated and sold without taking ticket holder name, email details.
Order is placed with these empty fields and tickets are generated and sold.

screenshot 12

The JSON response for such attendees is coming as-
```
{
	"id": 359,
	"order": {
		"id": 242,
		"identifier": "7ca7b854-57d2-4399-b3b3-68a3bee8b92f",
		"paid_via": "free",
		"amount": 0.0,
		"invoice_number": "O1489178012-242",
		"status": "completed",
		"completed_at": "2017-03-10T20:33:43",
		"payment_mode": null
	},
	"email": "",
	"lastname": "",
	"firstname": "",
	"ticket": {
		"id": 187,
		"price": 0.0,
		"description": "",
		"type": "free",
		"quantity": 100,
		"name": "My Ticket"
	},
	"checked_in": false
}
```

### PR DESCRIPTION
**Pull request by CosmicCoder96**
fixes #3358 

giphy 3

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
@niranjan94 
https://orega-server.herokuapp.com/e/941cfc9e/

you can try it out here once.

**Comment by CosmicCoder96** [comment]
@niranjan94 By any chance, did you purchased the tickets on a mobile ? I am unable to purchase tickets without entering a name as the button gets disabled if the form validation is not successful and I have used hover event as a trigger for validating, that might be the problem. If you can confirm this I can make necessary changes or if you were on desktop can you precisely tell me how exactly it happened. Thanks.

**Comment by niranjan94** [comment]
I was on desktop. I just clicked on the button as the page was loading. I suggest you do the validation in the server-side too.

**Comment by CosmicCoder96** [comment]
@niranjan94 I have made the server side changes too, now it won't create an order unless the fields have values in them , in case the js validation somehow fails, please review.

File: app/static/js/vendor/jquery/jquery.codezero.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ jQuery.fn.extend({
      */
     setFormLoading: function (isLoading, text) {
         return this.each(function () {
-            if (typeof flag === 'undefined') {
+            if (typeof isLoading === 'undefined') {
```
Comment (CosmicCoder96) [inline app/static/js/vendor/jquery/jquery.codezero.js]: flag was undefined, which broke the app when order initiation failed.

Comment (jack17529) [inline app/static/js/vendor/jquery/jquery.codezero.js]: @CosmicCoder96  Sir can I work on this issue

Comment (niranjan94) [inline app/static/js/vendor/jquery/jquery.codezero.js]: @jack17529 please browse through the issues at https://github.com/fossasia/open-event-orga-server/issues

File: app/helpers/ticketing.py [hunk 2]
```diff
@@ -373,6 +373,8 @@ def initiate_order_payment(form, paid_via=None):
                                              ticket_id=int(holders_ticket_ids[i]),
                                              email=holder_user.email,
                                              order_id=order.id)
+                if data['firstname'] == '' or data['lastname'] == '' or holder_user.email == '':
+                    return False
```
Comment (niranjan94) [inline app/helpers/ticketing.py]: @CosmicCoder96 please return a proper http header too. Use `HTTP 400 - Bad Request`

**Comment by CosmicCoder96** [comment]
@niranjan94 please review again,

**Comment by CosmicCoder96** [comment]
@niranjan94

**Pull request merged**